Framework/Kernel/src/ConfigService.cpp

```diff
@@ -325,6 +325,52 @@ void ConfigServiceImpl::setBaseDirectory() {
 #endif
 }
 
+namespace {
+// look for specific keys and throw an exception if one is found
+std::string checkForBadConfigOptions(const std::string &filename,
+                                     const std::string &propertiesString) {
+  std::stringstream stream(propertiesString);
+  std::stringstream resultPropertiesString;
+  std::string line;
+  int line_num = 0;
+  while (std::getline(stream, line)) {
+    line_num += 1; // increment early
+    bool is_ok = true;
+
+    // Check for common errors. Empty lines are ok, things that are a key
+    // without a value are a critical failure. Forbidden keys are just commented
+    // out.
+    if (line.empty() || (line[0] == '#')) {
+      // do nothing
+    } else if (line.find('=') == std::string::npos) {
+      std::stringstream msg;
+      msg << "Encontered invalid key \"" << line << "\"";
+      throw Exception::ParseError(msg.str(), filename, line_num);
+    } else if (line.find("FilterChannel") != std::string::npos) {
+      is_ok = false;
+    }
+
+    // Print warning to error channel and comment out offending line
+    if (!is_ok) {
+      const auto end = line.find("=");
+      std::cerr << "Encontered invalid key \"";
+      if (end != std::string::npos) {
+        std::cerr << Kernel::Strings::strip(line.substr(0, end));
+      } else {
+        std::cerr << Kernel::Strings::strip(line);
+      }
+      std::cerr << "\" in " << filename << " on line " << line_num << std::endl;
+
+      // comment out the property
+      resultPropertiesString << '#';
```

Comment:
This will disappear completely if I change the properties (i.e. setting default instrument from the gui). Not really an issue, just an observation.